src/pkg/egress/syslog/https_batch_test.go

```diff
@@ -83,37 +96,48 @@ var _ = Describe("HTTPS_batch", func() {
 	})
 
 	It("test batch dispatching with all logs in a given timeframe", func() {
-		env1 := buildLogEnvelope("APP", "1", "string to get log to 1024 characters:"+string_to_1024_chars, loggregator_v2.Log_OUT)
-		for i := 0; i < 10; i++ {
+		env1 := buildLogEnvelope("APP", "1", "short message", loggregator_v2.Log_OUT)
+		for i := 0; i < 5; i++ {
 			Expect(writer.Write(env1)).To(Succeed())
-			time.Sleep(99 * time.Millisecond)
 		}
-		Expect(drain.getMessagesSize()).Should(Equal(0))
-		time.Sleep(100 * time.Millisecond)
-		Expect(drain.getMessagesSize()).Should(Equal(10))
+		Expect(drain.getMessagesSize()).To(Equal(0))
+		Eventually(drain.getMessagesSize, 180*time.Millisecond).Should(Equal(5))
 	})
 
-	It("probabilistic test for race condition", func() {
-		env1 := buildLogEnvelope("APP", "1", "string to get log to 1024 characters:"+string_to_1024_chars, loggregator_v2.Log_OUT)
-		for i := 0; i < 10; i++ {
+	It("triggers multiple flushes when exceeding the batch", func() {
+		env := buildLogEnvelope("APP", "1", "string to get log to 400 characters:"+stringTo256Chars, loggregator_v2.Log_OUT)
+		// This
+		for i := 0; i < 15; i++ {
+			Expect(writer.Write(env)).To(Succeed())
+		}
+		// This is a less flaky approach to test for batch size triggered sends:
+		// with only Time based sends, it either takes
+		Eventually(drain.getMessagesSize, 180*time.Millisecond).Should(Equal(15))
```

Comment:
Can you please make this timeout a big longer to account for some processing time.